lib/engine/config.ex

```diff
@@ -32,10 +32,14 @@ defmodule Engine.Config do
   end
 
   @impl true
-  def sign_config(table_name \\ @table_signs, sign_id) do
-    case :ets.lookup(table_name, sign_id) do
-      [{^sign_id, config}] -> config
-      _ -> :off
+  def sign_config(table \\ @table_signs, sign)
+  def sign_config(table, %{id: id, default_mode: default}), do: lookup(table, id, default)
+  def sign_config(table, %{id: id}), do: lookup(table, id, :off)
```

Comment:
nit: There isn't an immediate need to implement this for bus signs, but generally we've been looking to nudge the rail and bus code closer together over time. So instead of having this be optional, we could hardcode a value of `:off` in the bus state, and then they could use the same code path.